controller/src/main/java/io/pravega/controller/server/rest/ModelHelper.java

```diff
@@ -70,6 +80,24 @@ public static final StreamConfiguration getCreateStreamConfig(final CreateStream
                 .build();
     }
 
+    private static RetentionPolicy getConsumptionRetentionPolicy(ConsumptionLimits consumptionLimits) {
+        RetentionPolicy.ConsumptionLimits.Type type;
+        int multiplier;
+        switch (consumptionLimits.getType()) {
+            case SIZE_MB:
+                type = RetentionPolicy.ConsumptionLimits.Type.SIZE_BYTES;
+                multiplier = BYTES_TO_MB;
+                break;
+            case TIME_MINUTES:
+                type = RetentionPolicy.ConsumptionLimits.Type.TIME_MILLIS;
+                multiplier = MILLIS_TO_MINUTES;
+                break;
+            default:
```

Comment:
ConsumptionLimit.UNKNOWN is not covered here. That is a valid usecase and should not throw "NotImplementedException"

Reply:
unknown, as explained, is not a valid use case. it is a protobuf enum evolution safety measure to allow for new values to be mapped to unknown. 

if we want NONE, that should be added separately. However, i think a limit with null on min or max or both is sufficient to provide that and we may not need a separate enum type NONE

Reply:
The design allows a user to skip specifying the max and min limits and so there should be an ENUM to represent that option, perhaps a NONE.

Reply:
Yes NONE is same as min=null and max=null so NONE can be skipped.